### PR TITLE
SQL-729: Add support for local mongohoused

### PR DESCRIPTION
**Pull request by henretty**
Updates run_adl.sh to allow using a local copy of mongohouse source. This is enabled with the `HAVE_LOCAL_MONGOHOUSE` and `LOCAL_MONGOHOUSE_DIR` environment variables. 

The only change to previous default behavior is that mqlrun and the mongosql library are downloaded on every run. This is a DevX change as it 1) guarantees that these tools are the newest available and 2) guarantees OS compatibility when e.g., using a mac and running run_adl.sh in a linux container with mongohouse source mounted from the mac.

### CODE REVIEW

**Review by nbagnard — APPROVED** [review]
LGTM!

File: resources/run_adl.sh [hunk 1]
```diff
@@ -168,34 +173,45 @@ check_mongohoused
...
   if [ $ARG = $START ]; then
     echo "Starting $MONGOHOUSED"
-    # Install and start mongohoused
-    git config --global url.git@github.com:.insteadOf https://github.com/
-    # Clone the mongohouse repo
-    if [ ! -d "$MONGOHOUSE_DIR" ]; then
-      git clone $MONGOHOUSE_URI
+    if [[ $HAVE_LOCAL_MONGOHOUSE -eq 1 ]]; then
+        if [ ! -d "$LOCAL_MONGOHOUSE_DIR" ]; then
+            echo "ERROR: LOCAL_MONGOHOUSE_DIR is not a directory"
```
Comment (bucaojit) [inline resources/run_adl.sh:178]: Let's print the value of the directory here
```suggestion
            echo "ERROR: $LOCAL_MONGOHOUSE_DIR is not a directory"
```

**Review by bucaojit — APPROVED** [review]
Looks good.  One minor suggestion.

**Comment by henretty** [comment]
evergreen merge

**Comment by henretty** [comment]
@bucaojit @nbagnard Another review requested as I added some non-trivial logic to restore `mongosql` and `libmongosql.a` to their original versions when running with a local mongohouse source tree. 

This is to ensure that we don't end up with binaries for an incompatible OS e.g., if a developer is on a Mac and they run the ADL tests on a Linux container.

This change only substantially affects behavior when run with `HAVE_LOCAL_MONGOHOUSE`.

**Review by bucaojit — APPROVED** [review]
Looks good.  One comment about the .orig files. 
No need to retag.

File: resources/run_adl.sh [hunk 2]
```diff
@@ -230,5 +253,19 @@ else
     MONGOHOUSED_PID=$(< $TMP_DIR/${MONGOHOUSED}.pid)
     echo "Stopping $MONGOHOUSED, pid $MONGOHOUSED_PID"
     pkill -TERM -P ${MONGOHOUSED_PID}
+    if [[ $HAVE_LOCAL_MONGOHOUSE -eq 1 && -d "$LOCAL_MONGOHOUSE_DIR" ]]; then
+        echo "Restoring ${TENANT_CONFIG}"
+        cd $LOCAL_MONGOHOUSE_DIR
+        cp ${TENANT_CONFIG}.orig ${TENANT_CONFIG}
```
[the remark this answers is not in the document]
Comment (henretty) [reply resources/run_adl.sh]: Done!

**Comment by henretty** [comment]
I don't have permissions to merge. @bucaojit can you do this?

**Pull request merged**